### PR TITLE
Refactor the Launch a Free Site test to not require a signup.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

- use existing LoginFlow to log in as existing user to run the test.
- use existing DeleteSiteFlow in place of DeletAccount flow for cleanup.
- add in path to handle Domain Upsell during site launch phase.

#### Testing instructions

Wait for CircleCI to complete, or pull the changes locally and run:

```
 mocha specs/wp-launch-spec.js
```

Example Runs (isolated to just this test):
- https://app.circleci.com/pipelines/github/Automattic/wp-e2e-tests-for-branches/59836/workflows/50ad9a79-c5b9-4703-a6d6-3dad3e331c6c/jobs/145369/parallel-runs/1?filterBy=ALL
- https://app.circleci.com/pipelines/github/Automattic/wp-e2e-tests-for-branches/59837/workflows/20231f2d-df5e-4e10-84a0-8ca879138180/jobs/145370/parallel-runs/1?filterBy=ALL

### CODE REVIEW

**Comment by worldomonation** [comment]
Recent run has failures on both desktop and mobile, but are unrelated failures (signup tests).

File: test/e2e/specs/wp-launch-spec.js [hunk 1]
```diff
@@ -42,32 +44,37 @@ before( async function () {
...
-// Potential issue that trigger this failure:
-// https://github.com/Automattic/wp-calypso/issues/50273
-describe.skip( `[${ host }] Launch (${ screenSize }) @signup @parallel`, function () {
+describe( `[${ host }] Launch (${ screenSize }) @signup @parallel`, function () {
 	this.timeout( mochaTimeOut );
 
 	describe( 'Launch a free site', function () {
-		const accountName = dataHelper.getNewBlogName();
 		const siteName = dataHelper.getNewBlogName();
+		console.log( siteName );
```
Comment (razvanpapadopol) [inline test/e2e/specs/wp-launch-spec.js]: Is there a reason to leave the `console.log` here when tests are running in CI?

**Review by razvanpapadopol — APPROVED** [review]
The changes are looking good 👌🏼  
Thanks for re-enabling the Launch test and for the newly added efficiency!

File: test/e2e/specs/wp-launch-spec.js (hunk 1 above)
Comment (worldomonation) [reply test/e2e/specs/wp-launch-spec.js]: That was my mistake. Fixing now.

As an aside, I see that we disable the linter rule for `console.log` under `test/e2e`. I believe it should be a warning, to catch instances like this.

Comment (worldomonation) [reply test/e2e/specs/wp-launch-spec.js]: Removed the line in subsequent revision.

**Pull request merged**